IBX-6878: Customize search sorting (#2226)
ibexa/documentation-developer: docs/administration/back_office/search_sorting.md

> ---

Reviewer: Change file name to "customize_search_sorting.md" to be consistent with what is visible in TOC.

Author: Done in 5527c36956be0e2296ad36fbdfb8b6386afec4dc

description: Add a "sort by" method to the Back Office search result page.
---

# Customize search sorting

> To add an entry to the "Sort by" to the Back Office search result page, create a service implementing the `SortingDefinitionProviderInterface` and tagged `ibexa.search.sorting_definition.provider`.

Reviewer 2: Let's use FQCN when we introduce symbol: `Ibexa\Contracts\Search\SortingDefinition\SortingDefintionProviderInterface`

Author: ```suggestion
To add an entry to the "Sort by" to the Back Office search result page, create a service implementing the `Ibexa\Contracts\Search\SortingDefinition\SortingDefintionProviderInterface` and tagged `ibexa.search.sorting_definition.provider`.
```

> To add an entry to the "Sort by" to the Back Office search result page, create a service to implement the `Ibexa\Contracts\Search\SortingDefinition\SortingDefintionProviderInterface` and tag `ibexa.search.sorting_definition.provider`.

Reviewer: Some introduction is missing here, like: "You can customize search sorting by adding an entry (...). To do it, (...)".

Author: @julitafalcondusza What about this?:
```suggestion
You can customize the **Sort by** menu in the Back Office search result page. To do so, create a service implementing the `Ibexa\Contracts\Search\SortingDefinition\SortingDefintionProviderInterface` and tag it `ibexa.search.sorting_definition.provider`.
```

> To add an entry to the "Sort by" to the Back Office search result page, create a service implementing the `Ibexa\Contracts\Search\SortingDefinition\SortingDefintionProviderInterface` and tagged `ibexa.search.sorting_definition.provider`.

Reviewer: ```suggestion
To add an entry to the "Sort by" to the Back Office search result page, create a service to implement the `Ibexa\Contracts\Search\SortingDefinition\SortingDefintionProviderInterface` and tag `ibexa.search.sorting_definition.provider`.
```

> The following example class implements `SortingDefinitionProviderInterface::getSortingDefinitions`, and add two definitions to sort by section name.

Reviewer: ```suggestion
The following example class implements `SortingDefinitionProviderInterface::getSortingDefinitions`, and adds two definitions to sort by section name.
```

> A sorting definition is an identifier, a menu label, a list of [Content Search's Sort Clauses](sort_clause_reference.md#sort-clauses), even [custom ones](create_custom_sort_clause.md), and a priority to position it in the menu.

Reviewer: ```suggestion
A sorting definition contains an identifier, a menu label, a list of [Content Search's Sort Clauses](sort_clause_reference.md#sort-clauses), including [custom ones](create_custom_sort_clause.md), and a priority to position it in the menu.
```

> For the menu label, its also implements `TranslationContainerInterface::getTranslationMessages` to provide two default English translations in the `ibexa_search` namespace.

Reviewer: ```suggestion
It also implements `TranslationContainerInterface::getTranslationMessages` to provide two default English translations in the `ibexa_search` namespace.
```

> This example is coded in `src/Search/SortingDefinition/Provider/SectionNameSortingDefinitionProvider.php`:

Reviewer: ```suggestion
The example below is coded in `src/Search/SortingDefinition/Provider/SectionNameSortingDefinitionProvider.php`:
```

``` php hl_lines="22"
[[= include_file('code_samples/back_office/search/src/Search/SortingDefinition/Provider/SectionNameSortingDefinitionProvider.php') =]]
```

> Its service definition is appended to `config/services.yaml`:

Reviewer: ```suggestion
The service definition is added to `config/services.yaml`:
```

``` yaml hl_lines="5"
[[= include_file('code_samples/back_office/search/config/append_to_services.yaml') =]]
```

Translation file can be extracted with a `translation:extract` command, such as `php bin/console translation:extract en --dir=src --output-dir=translations` to obtain an `translations/ibexa_search.en.xlf` file.

> Or translation file can be manually produced, like the following `translations/messages.en.yaml`:

Reviewer: ```suggestion
It can be also created manually, as `translations/messages.en.yaml` file below :
```
